File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/TokenRepository.java
package com.raxim.myscoutee.profile.repository.mongo;

import java.util.List;
import java.util.UUID;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

import com.raxim.myscoutee.profile.data.document.mongo.Token;

public interface TokenRepository extends MongoRepository<Token, UUID>{

    @Query("{deviceKey: ?0}")
    Token findByDeviceId(String deviceId);

    @Query("{uuid: { $in: ?0 }}")
    List<Token> findByUserIds(List<UUID> uuids);
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/repository/mongo/TokenRepository.kt
package com.raxim.myscoutee.profile.repository.mongo

import com.raxim.myscoutee.profile.data.document.mongo.Token
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import java.util.*

@RepositoryRestResource(
    collectionResourceRel = "tokens",
    path = "tokens"
)
interface TokenRepository : MongoRepository<Token, UUID>,
    QuerydslPredicateExecutor<Token> {

    @Query("{deviceKey: ?0}")
    fun findByDeviceId(deviceId: String): Token?

    @Query("{uuid: ?0}")
    fun findByUserIds(uuids: Array<UUID>): List<Token>
}